src/gps/GPS.cpp

```diff
@@ -646,7 +652,33 @@ bool GPS::setup()
                         }
                     }
                 }
-            } else {
+            } 
+
+            // M9
+            if (strncmp(info.hwVersion, "00190000", 8) == 0) {
+                LOG_INFO("Using defaults for config for UBLOX M9\n");
+                // reset, 0x00, 0x00, 0x00, // byte length
+                // manually reset in development
+                // TODO: add to python CLI
+                // msglen = makeUBXPacket(0xb5, 0x62, sizeof(_message_CFG_RST), _message_CFG_RST);
+                // _serial_gps->write(UBXscratch, msglen);
+                
+                // Turn off unwanted NMEA messages, set update rate
+                // --> THIS IS FAILING CURRENTLY
+                // --> NEEDS TRANSLATION TO 32.01
+                msglen = makeUBXPacket(0xb5, 0x62, sizeof(_message_1HZ), _message_1HZ);
+                _serial_gps->write(UBXscratch, msglen);
+
+                if (getACK(0x06, 0x08, 500) != GNSS_RESPONSE_OK) {
+                    LOG_WARN("Unable to set GPS update rate.\n");
+                } else {
+                    LOG_INFO("Successfully set GPS update rate.!\n");
+                }
+                delay(1000);
+            }
```

Comment:
What other messages should go here?  Not super familiar with all of these protocols.  A dump of the hex code for M9 would be helpful BUT, a pointer directly to config.  Would be helpful.

For instance, what does `_message_VALSET_DISABLE_NMEA_RAM` correspond to?  Would it be [3.10.13 UBX-CFG-NMEA (0x06 0x17)](https://content.u-blox.com/sites/default/files/u-blox-M9-SPG-4.04_InterfaceDescription_UBX-21022436.pdf)

Reply:
@GPSFan ^^...  cc: @lolsborn 

@GPSFan if you made comments in here with what configs our commands you think we absolutely need, we can grind them out.  We'd appreciate it! 🙏